src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -408,7 +409,7 @@ public static ChatDecorator getServerChatSubmittedDecorator()
                 return message; // Vanilla should never get here with the patches we use, but let's be safe with dumb mods
 
             return onServerChatSubmittedEvent(sender, getRawText(message), message);
-        });
+        }, Util.ioPool());
```

Comment:
I believe the background executor would be more appropriate here, as the IO pool should be reserved only for IO tasks.